### PR TITLE
[stable10] Combine then statements into one step 

### LINKED ISSUE #33499: acceptance test code difference between stable10 & master
`bootstrap/Sharing.php` is different in master & stable10 
code added in https://github.com/owncloud/core/pull/22750 is missing in master
```
				if (\substr($field, 0, 10) === "share_with") {
					$value = \str_replace(
						"REMOTE",
						$this->getRemoteBaseUrl(),
						$value
					);
					$value = \str_replace(
						"LOCAL",
						$this->getLocalBaseUrl(),
						$value
					);
				}
				if (\substr($field, 0, 6) === "remote") {
					$value = \str_replace(
						"REMOTE",
						$this->getRemoteBaseUrl() . '/',
						$value
					);
					$value = \str_replace(
						"LOCAL",
						$this->getLocalBaseUrl() . '/',
						$value
					);

### PR DESCRIPTION
**Pull request by individual-it**
backport of #33483
backport of #33504
fixes #33499

### CODE REVIEW

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
Conflicts resolved - ket's see what CI  says.

**Comment by phil-davis** [comment]
CI https://drone.owncloud.com/owncloud/core/12546/290
```
remote doesn't have value https://federated
```
Now to look at what happened to
```
Scenario: Remote sharee can see the pending share # /drone/src/tests/acceptance/features/apiFederation/federated.feature:53
```

File: tests/acceptance/features/bootstrap/Sharing.php [hunk 1]
```diff
@@ -1233,6 +1265,28 @@ public function checkShareFields($body) {
 						$value
 					);
 				}
+				if (!$this->isFieldInShareResponse($field, $value)) {
```
[the remark this answers is not in the document]
Comment (individual-it) [reply tests/acceptance/features/bootstrap/Sharing.php]: I don't think that is the issue. `$value` is set in the function earlier
and the problem is in `FeatureContext::checkFields()`

[the remark this answers is not in the document]
Comment (phil-davis) [reply tests/acceptance/features/bootstrap/Sharing.php]: I see - the lines of code above here are still doing the ``getRemoteBaseUrl()`` and ``getLocalBaseUrl()`` stuff.
And that code puts a ``/`` after the base URL for both remote and local.
In master that is now a call to ``replaceValuesFromTable``

**Review by phil-davis — APPROVED** [review]
Looks good now - let's see what CI says

**Pull request merged**